pkg/graph/resolvers/itgovactions/lcidactions/update_lcid.go

```diff
@@ -0,0 +1,85 @@
+package lcidactions
+
+import (
+	"time"
+
+	"github.com/guregu/null"
+
+	"github.com/cmsgov/easi-app/pkg/apperrors"
+	"github.com/cmsgov/easi-app/pkg/models"
+)
+
+// GetUpdateLCIDAction is a method to set all the relevant data fields for that are a result of an Update LCID action.
+func GetUpdateLCIDAction(
+	intake models.SystemIntake,
+	expirationDate *time.Time,
+	nextSteps *models.HTML,
+	scope *models.HTML,
+	costBaseline *string,
+	userInfo models.UserInfo,
+) models.Action {
+
+	action := getBaseLCIDAction(intake, expirationDate, nextSteps, scope, costBaseline, userInfo)
+	action.ActionType = models.ActionTypeUPDATELCID
+
+	return action
+
+}
+
+// IsLCIDValidToUpdate checks if you can update the LCID for an intake, if not, it will return an error
+// An LCID must have already been issued in order to be updateable
+func IsLCIDValidToUpdate(intake *models.SystemIntake) error {
+	if !intake.LifecycleID.Valid {
+		return &apperrors.BadRequestError{
+			Err: &apperrors.InvalidActionError{
+				ActionType: models.ActionTypeUPDATELCID,
+				Message:    "an LCID has not been issued for this intake, an LCID must be issued before it can be updated",
+			},
+		}
+	}
+	return nil
+}
+
+// getBaseLCIDAction returns the basic entries for an aciton. Action type is handled by the calling action
+func getBaseLCIDAction(intake models.SystemIntake,
+	expirationDate *time.Time,
+	nextSteps *models.HTML,
+	scope *models.HTML,
+	costBaseline *string,
+	userInfo models.UserInfo,
+) models.Action {
+
+	action := models.Action{
+		IntakeID:       &intake.ID,
+		ActorName:      userInfo.CommonName,
+		ActorEmail:     userInfo.Email,
+		ActorEUAUserID: userInfo.EuaUserID,
+		Step:           &intake.Step,
+
+		LCIDExpirationChangeNewDate:      intake.LifecycleExpiresAt,
+		LCIDExpirationChangePreviousDate: intake.LifecycleExpiresAt,
+
+		LCIDExpirationChangeNewScope:      intake.LifecycleScope,
+		LCIDExpirationChangePreviousScope: intake.LifecycleScope,
+
+		LCIDExpirationChangeNewNextSteps:      intake.DecisionNextSteps,
+		LCIDExpirationChangePreviousNextSteps: intake.DecisionNextSteps,
+
+		LCIDExpirationChangeNewCostBaseline:      intake.LifecycleCostBaseline,
+		LCIDExpirationChangePreviousCostBaseline: intake.LifecycleCostBaseline,
+	}
+	//Action is set based on if the value was set or not. If not set, the new value and the old are the same
```

Comment:
* This comment should probably be a bit higher up, above where all the "Previous" and "New" fields get initialized, to clarify why the fields are getting initialized with the same value.
* "Action" isn't what's getting set here; it's the various fields on the action.
* It might be worth adding that it's an intentional choice that the "Previous" and "New" versions of a field might have the same value if that field's not updated, to preserve the existing behavior in `NewCreateActionExtendLifecycleID()`.